app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt

```diff
@@ -411,12 +416,40 @@ class WebRenderFragment : EngineViewLifecycleFragment(), Session.Observer {
         if (!browserOverlay.isVisible && session.isYoutubeTV &&
                 event.keyCode == KeyEvent.KEYCODE_BACK) {
             val escKeyEvent = KeyEvent(event.action, KeyEvent.KEYCODE_ESCAPE)
-            (activity as MainActivity).dispatchKeyEvent(escKeyEvent)
+
+            /**
+             * If YouTube guide-list element is focused:
+             * - Suppress the DOWN,BACK key event
+             * - Change the UP,BACK key event to go back in history before YouTube
+             * Else:
+             * - Dispatch ESC key event
+             */
+            val jsCallback = ValueCallback<String> {
+                if (it == "true") {
+                    if (event.action == KeyEvent.ACTION_DOWN) Unit
+                    else goBackBeforeYouTube()
+                } else (activity as MainActivity).dispatchKeyEvent(escKeyEvent)
+            }
+            // This will return true if the currently focused YouTube element is a button in the left sidebar, false otherwise
+            webView?.evalJS("""
+                (function () {
+                    return document.activeElement.parentElement.parentElement.id === 'guide-list';
+                })();
+                """,
+                    jsCallback)
             return true
         }
         return false
     }
 
+    private fun goBackBeforeYouTube() {
+        val webView = (webView as ViewGroup).getChildAt(0) as WebView
+        val backForwardUrlList = webView.copyBackForwardList().toList().map { it.originalUrl }
+        val youtubeIndex = backForwardUrlList.lastIndexOf(YOUTUBE_TILE_URL)
```

Comment:
Finding the index and going back the appropriate number of steps would be a good candidate for writing tests, e.g.:
- How does this handle youtube.com/tv URLs?
- What happens if the CDN url is not in the format we expect (no trailing slash)?
- What happens if the CDN is at the end of the url list?
- What happens if the CDN url is not in the list?
- What if the CDN url is first?
- What if we remove firefox:home from the pages loaded by the application?

You get the idea. :P